pkg/queue/manager.go

```diff
@@ -341,19 +341,20 @@ func (m *Manager) deleteWorkloadFromQueueAndClusterQueue(w *kueue.Workload, qKey
 	}
 }
 
-// QueueAssociatedInadmissibleWorkloads moves all associated workloads from
-// inadmissibleWorkloads to heap. If at least one workload is moved,
-// returns true. Otherwise returns false.
-func (m *Manager) QueueAssociatedInadmissibleWorkloads(ctx context.Context, w *kueue.Workload) {
+// QueueAssociatedInadmissibleWorkloadsAfter moves all associated workloads from
+// inadmissibleWorkloads to the heaps, after executing action, while holding the
+// lock. If at least one workload is moved, returns true. Otherwise returns false.
+func (m *Manager) QueueAssociatedInadmissibleWorkloadsAfter(ctx context.Context, w *kueue.Workload, action func()) {
```

Comment:
You changed the semantics of this function. Add a comment that requeueing happens only if the workload triggering it is an admitted one?

Reply:
In theory this is not changing the behavior, as we would only call this function for admitted workloads. So it's more of a code simplification.
But good thing to note in the documentation.